src/compiler.jl

```diff
@@ -3252,7 +3252,7 @@ function create_abi_wrapper(enzymefn::LLVM.Function, TT, rettype, actualRetType,
     if is_adjoint && rettype <: Active
         @assert !sret_union
         if allocatedinline(actualRetType) != allocatedinline(literal_rt)
-            @show actualRetType, literal_rt, rettype
+            @safe_show actualRetType, literal_rt, rettype
```

Comment:
I would recommend turning this into a better `throw(AssertionError(string))` and printing these directly into the string, and getting rid o the assert below

Reply:
I turned it into an `AssertionError` and replaced a few other occurrences of the same pattern in src/compiler.jl as well.

Reply:
Awesome!